src/hooks/useRegistrationReducer.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  RegistrationReducerAction,
+  RegistrationReducerData,
+  RegistrationReducerDataItem,
+  SelectedItemProperties,
+} from '@app/components/pages/profile/[name]/registration/types'
+import { useLocalStorageReducer } from '@app/hooks/useLocalStorage'
+
+const randomSecret = () => {
+  const bytes = Buffer.allocUnsafe(32)
+  return `0x${window.crypto.getRandomValues(bytes).toString('hex')}`
+}
+
+const defaultData: RegistrationReducerDataItem = {
+  stepIndex: 0,
+  queue: ['pricing', 'info', 'transactions', 'complete'],
+  years: 1,
+  reverseRecord: false,
+  records: {},
+  resolver: '',
+  permissions: {},
+  secret: '',
+  started: false,
+  address: '',
+  name: '',
+}
+
+const isBrowser = !!(
+  typeof window !== 'undefined' &&
+  window.document &&
+  window.document.createElement
+)
+
+const makeDefaultData = (selected: SelectedItemProperties): RegistrationReducerDataItem => ({
+  stepIndex: 0,
+  queue: ['pricing', 'info', 'transactions', 'complete'],
+  years: 1,
+  reverseRecord: false,
+  records: {},
+  resolver: '',
+  permissions: {},
+  secret: randomSecret(),
+  started: false,
+  ...selected,
+})
+
+export const getSelectedIndex = (
+  state: RegistrationReducerData,
+  selected: SelectedItemProperties,
+) => state.items.findIndex((x) => x.address === selected.address && x.name === selected.name)
+
+/* eslint-disable no-param-reassign */
+const reducer = (state: RegistrationReducerData, action: RegistrationReducerAction) => {
+  let selectedItemInx = getSelectedIndex(state, action.selected)
+
+  if (!isBrowser) return
```

Comment:
What situation would there not be a browser? Is this for unit tests?

Reply:
nextjs rendering